### PR TITLE
docs: Add share p2p documentation

### PR DESCRIPTION
**Pull request by derrandz**
## Overview

Self-explanatory

## How to Test Changes

Launch:

```
$ godoc -http=:6060
```
And browse the impacted packages.

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

**Comment by ramin** [comment]
@renaynay on a quick glance are these docs still representative of what they document? if so, I'll take this over and try and bring it home

**Comment by derrandz** [comment]
HI @ramin these docs should still be accurate, but would require updates once https://github.com/celestiaorg/celestia-node/issues/2604 is closed as well after all of the pruning and sharding logic happens as it will impact peer manager full nodes discovery and should be documented then.

**Comment by ramin** [comment]
@derrandz if you fix these too long line lint errors quickly i'll merge this. otherwise i'll close out and ill add more documentation as we get closer to mainnet in coming month

**Comment by derrandz** [comment]
Cool, will fix and let's merge

**Comment by derrandz** [comment]
@ramin made a few corrections to the docs and made the linter happy. Should be ready to go

File: cmd/celestia/rpc.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ import (
 	"github.com/celestiaorg/celestia-node/state"
 )
 
-const authEnvKey = "CELESTIA_NODE_AUTH_TOKEN" //nolint:gosec
```
Comment (ramin) [inline cmd/celestia/rpc.go:26]: @distractedm1nd thinkgs that that this is causing the continued linter failures, lets add back and see if we can get a pass @derrandz

**Review by ramin — CHANGES_REQUESTED** [review]

File: cmd/celestia/rpc.go (hunk 1 above)
Comment (derrandz) [reply cmd/celestia/rpc.go]: true. Removed it due to linter complaining locally, dunno how to have the local linter also detected gosec stuff but we should do it.

**Review by distractedm1nd — APPROVED** [review]

**Comment by distractedm1nd** [comment]
Let's merge this and we can make edits as necessary to the contents, this PR is getting stale

**Pull request merged**